x-pack/legacy/plugins/infra/server/routes/snapshot/index.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { InfraBackendLibs } from '../../lib/infra_types';
+import { InfraSnapshotRequestOptions } from '../../lib/snapshot';
+import { UsageCollector } from '../../usage/usage_collector';
+import { parseFilterQuery } from '../../utils/serialized_query';
+import { InfraSnapshotNode, SnapshotRequest } from './types';
+
+export const initSnapshotRoute = (libs: InfraBackendLibs) => {
+  const { framework } = libs;
+
+  framework.registerRoute<SnapshotRequest, Promise<{ nodes: InfraSnapshotNode[] }>>({
+    method: 'POST',
+    path: '/api/metrics/snapshot',
+    handler: async req => {
+      const source = await libs.sources.getSourceConfiguration(req, req.payload.sourceId);
```

Comment:
See the note about adding an IO-TS type for validating the `payload.request` object for the node details api.